### PR TITLE
[receiver/solacereceiver] Solace Receiver (#10572)

### PR DESCRIPTION
**Pull request by mcardy**
**Description:** Add Solace receiver to receive trace data from a [Solace PubSub+ Event Broker](https://solace.com/products/event-broker/)

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10572

**Testing:** Full test coverage using mocks

**Documentation:** Component documented in receiver's README.md

Sponsor: @djaglowski

### CODE REVIEW

**Comment by djaglowski** [comment]
@mcardy, please sign the CLA

File: receiver/solacereceiver/config.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		return errMissingAuthDetails
+	}
+	if len(strings.TrimSpace(cfg.Queue)) == 0 {
+		return errMissingQueueName
+	}
+	return nil
+}
+
+type Transport struct {
+	TLS configtls.TLSClientSetting `mapstructure:"tls,omitempty"`
```
Comment (djaglowski) [inline receiver/solacereceiver/config.go:66]: Is it necessary to nest this under `transport`? The vast majority of components using this config struct present `tls` at the top level of the config.

**Review by djaglowski — COMMENTED** [review]
Overall, the code looks very clean and complete. I left just one comment. 

I'll trigger CI now, but the CLA check will be a blocker either way.

File: receiver/solacereceiver/config.go (hunk 1 above)
Comment (mgevantmakher) [reply receiver/solacereceiver/config.go:66]: we will present tls at the top level of the config

**Comment by mcardy** [comment]
I've changed `transport.tls.*` to `tls.*`, thanks for the review. We're in the process of assigning the CLA Manager internally for this and will have the CLA signed soon.

**Comment by mcardy** [comment]
@djaglowski could you please run the checks for the latest commit and review the latest changes? We have no more expected changes for the time being.

File: receiver/solacereceiver/receiver.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		retryTimeout:      1 * time.Second,
+		terminating:       atomic.NewBool(false),
+	}, nil
+}
+
+// Start implements component.Receiver::Start
+func (s *solaceTracesReceiver) Start(_ context.Context, _ component.Host) error {
+	recordReceiverStatus(receiverStateStarting)
+	var cancelableContext context.Context
+	cancelableContext, s.cancel = context.WithCancel(context.Background())
```
[the remark this answers is not in the document]
Comment (djaglowski) [reply receiver/solacereceiver/receiver.go:88]: Great, thanks for explaining to me

**Review by djaglowski — APPROVED** [review]

**Pull request merged**